### PR TITLE
Use Array.splice instead of Array.slice & Array.concat

### PR DESCRIPTION
**Pull request by yuler**

### CODE REVIEW

File: src/radix-trie.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ export class Leaf<T> {
   delete(value: T): boolean {
     const index = this.children.indexOf(value)
     if (index === -1) return false
-    this.children = this.children.slice(0, index).concat(this.children.slice(index + 1))
+    this.children.splice(index, 1)
```
Comment (koddsson) [inline src/radix-trie.ts:12]: Why this change? I'd personally not want to change this to mutate `this.children`. If we wanted to simplify this single line I'd like for it to be something like:

```suggestion
    this.children = [...this.children.slice(0, index), ...this.children.slice(index + 1))]
```

Comment (yuler) [reply src/radix-trie.ts:12]: I just want to simplify the line

Is `splice` the side effects of this affect other places?
